proto/spire/api/README.md

```diff
@@ -0,0 +1,11 @@
+# API protos
```

Comment:
We (maintainers) huddled and discussed this PR. I think we can merge without the README. I'll probably open up a follow up PR that moves the node and registration APIs somewhere else (without changing their proto package declaration).